enhancements/sha-bundle-digest.md

```diff
@@ -0,0 +1,80 @@
+-   Allow users to specify additional images to resolve digests via an environment variable.
+-   Add the ability to resolve image digests from image tag dynamically using local docker registry.
+-   OPEN QUESTION: Store digest and tag mappings in file so that user-written code can resolve tags to digest.
+
+### Non-Goals
+
+-   Migrate existing `operator-sdk` based operators from image tags to image digests.
+-   Identify locations in user code that reference images by tag.
+
+## Proposal
+
+-   Add `--image-digests` flag to `make bundle` in order to enable this feature to be run. 
+
+-   During bundle generation, discover all environment variables that begin with `RELATED_IMAGE_` then resolve the tags to digests. These variables should be set in `config/manager/manager.yaml` as suggested in [https://master.sdk.operatorframework.io/docs/best-practices/common-recommendation/#other-common-suggestions](https://master.sdk.operatorframework.io/docs/best-practices/common-recommendation/#other-common-suggestions). 
+
+
+### Risks and Mitigations
+
+Since the feature is opt-in, it carries the risk of the end-user enabling it without a correct understanding of the feature and its implications. This can be mitigated with clear documentation both on the website and in the CLI help section.
+
+## Design Details
+
+ -   Add a `--image-digests` flag to the `operator-sdk generate bundle` command that will enable this functionality.
+ -   Add a variable to the generated `Makefile` called `USE_IMAGE_DIGESTS` that defaults to `false` and when enabled, it passes the `--use-image-digests` flag to `operator-sdk generate bundle`.
+ -   When `operator-sdk generate-bundle` is called with the `--use-image-digests` flag enabled, it will execute the following logic.
+ 	1. Search the environment that is defined in the `manager.yaml` file for all variables that begin with `RELATED_IMAGE_` to build a list of images that need to have tags resolved to digests along with the Operator’s main image and rbac-proxy image.
+ 	 1.  For each image in the list, it will use `operator-manifest-tools` [https://github.com/operator-framework/operator-manifest-tools](https://github.com/operator-framework/operator-manifest-tools) to resolve the image tag to a digest
+ 	 1.  When building the cluster CSV, all images that have been identified as needing a digest instead of a tag should have that  digest as an annotation in the CSV.
+
+### Test Plan
+
+The resulting artifact of this EP will be a cluster CSV with image digests instead of tags in the image definitions as well as digest annotations. This feature will need an integration test that veifies that the knobs we provide to enable and configure image digests generates an expected CSV. The expected CSV will be manually created based on the expectations for how the feature should work. We can use this CSV in existing Operator Registry testing setups to verify that OPM correctly renders it.
```

Comment:
Can we put an example CSV for the Memcached demo operator here? It will help with visualization. Then the engineer who implements will know exactly what the output should look like.